File: Python/major.py
class Major: # Deklarasi class Major
    __majorName = "" # Untuk menyimpan nama jurusan, atribut berjenis private
    __code = "" # Untuk menyimpan kode jurusan, atribut berjenis private 
    __courseObject = [] # Untuk menyimpan objek kelas Course, atribut berjenis private

    def __init__(self, majorName = "", code = "", courseName = []): # Membuat konstruktor
        self.__majorName = majorName # Menginisialisasi atribut __majorName
        self.__code = code # Menginisialisasi atribut __code
        self.__courseObject = courseName # Menginisialisasi atribut __courseObject

    def setMajorName(self, majorName): # Setter untuk atribut __majorName
        self.__majorName = majorName # Menginisialisasi atribut __majorName

    def setCode(self, code): # Setter untuk atribut __code
        self.__code = code # Menginisialisasi atribut __code
   
    def getMajorName(self): # Getter untuk atribut __majorName
        return self.__majorName # Mengembalikan nilai atribut __majorName

    def getCode(self): # Getter untuk atribut __code
        return self.__code # Mengembalikan nilai atribut __code
    
    def getCourse(self): # Getter untuk atribut __courseObject
        return self.__courseObject # Mengembalikan nilai atribut __courseObject
    
    def getCourseIndex(self, index): # Getter untuk atribut __courseObject
        return self.__courseObject[index] # Mengembalikan nilai atribut __courseObject

File: C++/Major.cpp
// Import library yang dibutuhkan
#include <bits/stdc++.h>

using namespace std; // Menggunakan standard namespace

class Major // Deklarasi class Major
{
private:                       // Membuat atribut berjenis private
    string majorName;          // Untuk menyimpan nama jurusan
    string code;               // Untuk menyimpan kode jurusan
    list<Course> courseObject; // Untuk menyimpan objek dari class Course

public:
    /* Konstruktor */
    Major() // Membuat konstruktor tanpa isian
    {
        this->majorName = "";
        this->code = "";
    }

    Major(string majorName, string code, list<Course> course) // Membuat konstruktor dengan isian dari parameter
    {
        this->majorName = majorName;
        this->code = code;
        this->courseObject = course;
    }

    /* Setter dan Getter untuk setiap atribut di dalam class Major */
    void setMajorName(string majorName)
    {
        this->majorName = majorName;
    }

    void setCode(string code)
    {
        this->code = code;
    }

    string getMajorName()
    {
        return this->majorName;
    }

    string getCode()
    {
        return this->code;
    }

    list<Course> getCourse()
    {
        return this->courseObject;
    }

    /* Destruktor */
    ~Major() // Membuat destruktor untuk menghapus semua objek yang telah dibuat
    {
    }
};